src/sound/functions.hpp

```diff
@@ -0,0 +1,60 @@
+#pragma once
+#include "external.hpp"
+
+
+/// Function to finding a shared pointer object (spt) in vector (vec)
+/// 
+/// @param vec Vector where we want to find
+/// @param spt Shared pointer object to finded
+template <typename T>
+auto findInVector(std::vector< std::weak_ptr<T>>& vec,const std::shared_ptr<T>& spt) {
+	return std::find_if(vec.begin(), vec.end(),
+		[&spt](const std::weak_ptr<T>& wp) {
+			return !wp.expired() && wp.lock() == spt;
+		});
+}
+
+
+/// Function to finding a weak pointer object (spt) in vector (vec)
+/// 
+/// @param vec Vector where we want to find
+/// @param wptr Weak pointer object to finded
+template <typename T>
+auto findInVector(std::vector< std::weak_ptr<T>>& vec,const std::weak_ptr<T>& wptr) {
+	return std::find_if(vec.begin(), vec.end(),
+		[&wptr](const std::weak_ptr<T>& wp) {
+			return !wp.expired() && wp.lock() == wptr.lock();

```

Comment:
Remove `!wp.expired() &&` it's unnecessary
Also move the `wptr.lock()` outside the lambda or rewrite it like this:
```c++
[shptr = wptr.lock()] (const std::weak_ptr<T>& wp) {
	return wp.lock() == shptr;
}
```